pkg/space/space.go

```diff
@@ -29,6 +32,42 @@ func NewSpace(userSignup *toolchainv1alpha1.UserSignup, targetClusterName string
 	return space
 }
 
+// NewSpaceWithFeatureToggles is the same as NewSpace() but also does a feature toggle lottery drawing
+// and adds the corresponding feature annotations for features which "won" and should be enabled for the space.
+func NewSpaceWithFeatureToggles(userSignup *toolchainv1alpha1.UserSignup, targetClusterName string, compliantUserName, tier string, config toolchainconfig.ToolchainConfig) *toolchainv1alpha1.Space {
+	s := NewSpace(userSignup, targetClusterName, compliantUserName, tier)
+	addFeatureToggles(s, config)
+	return s
+}
+
+// addFeatureToggles does "lottery" drawing for all feature toggles defined in the configuration according to their weights.
+// And it adds the corresponding feature annotation to the space for features that won and should be enabled for the space.
+func addFeatureToggles(space *toolchainv1alpha1.Space, config toolchainconfig.ToolchainConfig) {
+	toggles := config.Tiers().FeatureToggles()
+	var winners string
+	for _, t := range toggles {
+		weight := int(t.Weight())
```

Comment:
the logic looks good, but `weight` is a bit confusing, could we use something like `threshold` instead?
(I believe `weight` would be used in case of load balancers, to distribute the load between targets)

Reply:
But it's not a threshold. It's a weight :)

Term "weight" is wildly used when there is a need of influence the probability of choosing a specific path / making a decision.

Reply:
okay... that's a matter of semantic, I'm not totally convinced by the "weight" term. As you said, it's used to choose a path (in load balancers, for example). Here, it looks like it's a minimal _score_ to get the feature enabled.
But naming is hard, I won't block the PR for this somehow minor point.

Reply:
+1 for weight from my side. As @alexeykazakov said, "weight" is widely used to influence the "importance" of some item in some list (i.e. weighted average, etc.).

Reply:
again, I believe that "weight" is used when there's a choice between multiple paths, but it's fine, let's not waste too much time on this term.